GetStream/stream-chat-android
# [3135] Allow passing long click handler to `MessageText`

### 🎯 Goal

Currently long pressing on any part of the text containing a link message will block it from opening message options. This is because 'Modifier.pointerInput()' inside `ClickableText` (foundation component) consumes all pointer events and does not back propagate them to the `Composable`s parent.

### 🛠 Implementation details

Trying to not consume all pointer events required too much low level re-writting so the following was implemented:

- Created a modified copy of `ClickableText` containing a long press handler
- Propagated a lambda to the previously mentioned long press handler that opens message options

### 🎨 UI Changes

**No UI changes**

### 🧪 Testing

Either find the long message containing both text and a link in the channel "Unknown Base" or create one of your own. Template should be plain text message + link.

- Launch Compose Sample App from the main or develop branch and long press on any part of the text of said message
- Launch Compose Sample App from the branch in this PR and long press on any part of the text of said message

Expected results: main/ develop branch versions will not open up message options while the version of the sample app from this branch will

### ☑️Contributor Checklist

#### General

#### Code & documentation
- [ ] ~~Comparison screenshots added for visual changes~~ (no visual changes)

### ☑️Reviewer Checklist

### 🎉 GIF

![options](https://user-images.githubusercontent.com/37080097/155704611-7b24d328-bd6e-4852-8590-02bcfdf261d8.gif)
ll request_
